In [2]:
import sys
import numpy as np

#### Ejercicio 1
¿Cuál es el tamaño en bytes de un array de 1000 elementos de tipo booleano?

In [7]:
array = np.random.rand(1000) > 0.5
sys.getsizeof(array)

1112

R:// 1112 bytes

#### Ejercicio 2
Cree un array cuyos elementos sean los enteros pares en  [1,100] y en orden decreciente. Muestre los 10 últimos por pantalla

In [10]:
array = np.sort(np.arange(2, 101, 2))[::-1]
array[len(array) - 11: len(array) - 1]

array([22, 20, 18, 16, 14, 12, 10,  8,  6,  4])

#### Ejercicio 3
Dado el array

2 4 5 6
0 3 7 4
8 8 5 2
1 5 6 1

Seleccione con una instrucción el subarray de elementos.

0 3 7 4

Después, seleccione el subarray de elementos.

2 5
8 5

In [11]:
array = np.array([[2, 4, 5, 6], [0, 3, 7, 4], [8, 8, 5, 2], [1, 5, 6, 1]])
array

array([[2, 4, 5, 6],
       [0, 3, 7, 4],
       [8, 8, 5, 2],
       [1, 5, 6, 1]])

In [12]:
array[1]

array([0, 3, 7, 4])

In [34]:
array[0:3:2, 0:3:2]

array([[2, 5],
       [8, 5]])

#### Ejercicio 4
Dados los arrays a = [1, 4, 2, 7] y b = [1, 3, 2, 9], 
obtenga la media aritmética de la diferencia (a-b).

In [35]:
a = np.array([1, 4, 2, 7])
b = np.array([1, 3, 2, 9])

In [37]:
r = a - b
r.mean()

-0.25

#### Ejercicio 5
Tengo valores de cordenadas (x, y) en las columnas del array

1.33 4.5

30.0 10.7

70.2 0.5

Agregue a este array las coordenadas (37.1, -3.6). Muestre en pantalla las dimensiones del nuevo array.

In [44]:
array = np.array([[1.33, 4.5], [30.0, 10.7], [70.2, 0.5]])
array

array([[ 1.33,  4.5 ],
       [30.  , 10.7 ],
       [70.2 ,  0.5 ]])

In [45]:
array = np.concatenate((array, [[37.1, -3.6]]), axis=0)
array.shape

(4, 2)

#### Ejercicio 6
Copie el array del ejercicio anterior. Traspóngalo (. Agregue ahora dos nuevos pares de coordenadas: (10.8, 3.0) y (35.8, 12.0).

In [52]:
array = np.array([[1.33, 4.5], [30.0, 10.7], [70.2, 0.5]])
array = np.transpose(array)
array

array([[ 1.33, 30.  , 70.2 ],
       [ 4.5 , 10.7 ,  0.5 ]])

In [53]:
array = np.concatenate((array, np.array([[10.8, 3.0]]).reshape(2,1) ), axis=1)
array = np.concatenate((array, np.array([[35.8, 12.0]]).reshape(2,1) ), axis=1)
array

array([[ 1.33, 30.  , 70.2 , 10.8 , 35.8 ],
       [ 4.5 , 10.7 ,  0.5 ,  3.  , 12.  ]])